### PR TITLE
Cleaned up the scraper cog

### PR DESCRIPTION
**Pull request by NotLeet**
I also added a little leeway to the requirements file.

I still plan on doing some stuff to hopefully clean up the way cog imports work, as I don't feel the current setup is ideal. I still think it's better than what we had though.

### CODE REVIEW

**Comment by RascalTwo** [comment]
Looking nice so far @NotLeet :+1:

**Comment by RascalTwo** [comment]
I saw one thing, the recursive cog finding code:

```Python

modules = []
for root, _, filenames in os.walk('cogs'):
    for filename in filenames:
        full_name = os.path.join(root, filename)
        if 'pycache' not in full_name and full_name.endswith('.py'):
            modules.append(full_name.replace('/', '.').replace('\\', '.')[0:-3])
return modules
```

When I run this, it returns a few `.py` files that are not cogs, specifically the utility files. Adding `'util' not in full_name` to the if statement solves this though.

Lastly, do you want to attempt to rebase your branch down to one or maybe two commits? Squashing the last three:

Fixing pylint errors&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;b3b54a2
Fixing more pylint errors&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0f17462
os.walk really needs that third...&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;53e05f8

should be enough.

[This](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) explains squashing pretty well. I can explain it to you directly if you want.

**Comment by NotLeet** [comment]
I kind of wanted to avoid making global decisions for the group about how granular our commits should be, or what should be able to constitute a module.

However, both of those changes seem really sensible, and if you're on board and no one else objects, I'll make them when I'm back at my computer and then merge the pull request.

**Comment by RascalTwo** [comment]
I do see your point,  only mentioned squashing to you because I made a push with adjustment-commits like yours, and was told I should've squashed them.

Also, what did you mean by `what should be able to constitute a module`?

I see this pull request as two things: a re-model/framework-creation for the Scraper cog, and adjustments to `requirements.txt` and `.gitignore`. Otherwords - basically just as your first commit - cleaning up the scraper cog into something more uniform, which is going to make scraping easier to do and more efficient.

File: bot.py [hunk 1]
```diff
@@ -18,14 +18,15 @@
...
+# Slightly longer method to find cogs recursively in the cogs folder
 def list_cogs():
-    return ['cogs.' + i.replace('/', '\\').split('\\')[0][:-3]
-            for i in os.listdir('cogs') if i.endswith('.py')]
-
+    cogs = []
+    for root, _, filenames in os.walk('cogs'):
+        for filename in filenames:
+            full_name = os.path.join(root, filename)
+            if full_name.endswith('.py') and all(word not in full_name for word in ['pycache', "__init__", "util"]):
```
Comment (raylu) [inline bot.py]: this line is too dense, mixes single- and double-quotes, and the `all` is unnecessary anyway

```python
if filename in ['pycache', '__init__', 'util']:
    continue
```

File: cogs/scraper/reddit_scraper.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        score = post['score']
+        author = post['author']
+        is_nsfw = post['over_18']
+        link = post['url']
+        title = post['title']
+
+        timediff = datetime.now() - datetime.fromtimestamp(post['created_utc'])
+        days = timediff.days
+        hours = timediff.seconds // 3600
+        minutes = (timediff.seconds // 60) % 60
```
Comment (raylu) [inline cogs/scraper/reddit_scraper.py]: 1. you probably want to compare to `datetime.utcnow()`
1. [timedelta](https://docs.python.org/3/library/datetime.html#timedelta-objects)'s have `hours` and `minutes`

File: cogs/scraper/reddit_scraper.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        if hours != 0:
+            if hours > 1:
+                created.append(str(hours) + ' hours')
+            else:
+                created.append(str(hours) + ' hour')
+        if minutes != 0:
+            if minutes > 1:
+                created.append(str(minutes) + ' minutes')
+            else:
+                created.append(str(minutes) + ' minute')
```
Comment (raylu) [inline cogs/scraper/reddit_scraper.py]: this seems like an obvious candidate for refactoring

File: cogs/scraper/reddit_scraper.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    async def get_subreddit_json(self, session, subreddit, category):
+        return await self.fetch_json(
+            session=session, url='https://www.reddit.com/r/' + subreddit + '/' + category + '/.json')
+
+    async def get_subreddit_top(self, session, subreddit, num_posts, category):
+        try:
+            sr_data = await self.get_subreddit_json(session=session, subreddit=subreddit, category=category)
+            sr_posts = sr_data['data']['children']
+        except KeyError:
+            return ['Posts could not be loaded, are you sure thats a subreddit?']
```
Comment (raylu) [inline cogs/scraper/reddit_scraper.py]: Posts could not be loaded. Are you sure that's a subreddit?

**Pull request merged**